nvmf-autoconnect/systemd/nvmf-connect-nbft.service.in

```diff
@@ -0,0 +1,14 @@
+# This unit is meant to be started by network management software
+# after a network interface defined in the NBFT gets set up
+[Unit]
+Description=Connect NBFT-defined NVMe-oF subsystems automatically
+ConditionPathExists=|/sys/firmware/acpi/tables/NBFT
+ConditionPathExists=|/sys/firmware/acpi/tables/NBFT1
+Wants=modprobe@nvme_fabrics.service
+After=modprobe@nvme_fabrics.service
+After=network-online.target
+Before=remote-fs-pre.target
+
+[Service]
+Type=oneshot
+ExecStart=@SBINDIR@/nvme connect-nbft
```

Comment:
Has any one tested this with nvme-fc boot?  We also support nvme-fc boot too.

Reply:
This command can be issued multiple times with no risk. The kernel notices if some of the requested connections are already established.

The point here is that during initrd processing, not all connections may have been brought up successfully, because not all network interface have been up. The typical usage scenario is multipath, where we don't want to get stuck in the initrd because one path is not available (yet). Thus  dracut will proceed mounting and switching root as soon as a single path is available for accessing the root FS. This, in turn, may cause dracut to switch "prematurely" if there are additional healthy paths, i.e. before it has brought up paths that can be brought up. For that reason, we need to make a 2nd bring-up attempt after switching root.

This way, we avoid being stuck in the initrd if paths are missing, while still bringing up all paths if all is good.

Reply:
@johnmeneghini's comment makes me realize that this is the legacy syntax. It must be `nvme connect-all --nbft` for upstream.

Reply:
Updated the nvme invocation. Anything else to do?

Reply:
I still have some concerns about how this will work together with dracut and network manager during  nbft boot. I've asked  @tbzatek to review this.  Can we add Tomas as a reviewer?

https://github.com/tbzatek

I'd also like to test these changes in my nvmft boot testbed before we merge. I'll do that next week.